collect_app/src/main/java/org/odk/collect/android/tasks/SaveFormToDisk.java

```diff
@@ -212,9 +210,9 @@ private void updateInstanceDatabase(boolean incomplete, boolean canEditAfterComp
             InstancesRepository instances = new DatabaseInstancesRepository();
             Instance instance = instances.getOneByPath(instancePath);
             if (instance != null) {
-                uri = Uri.withAppendedPath(InstanceColumns.CONTENT_URI, instance.getId().toString());
+                Uri instanceUri = Uri.withAppendedPath(InstanceColumns.CONTENT_URI, instance.getId().toString());
```

Comment:
This should not have been reusing the field as it's just used locally. One idea I have for why this crash happens is that somehow there is an instance that matches the path in the DB but the update fails. Then the else branch below is hit but the uri is set to the instance uri rather than the form uri as it should be. This feels like a really safe fix to me but it would be good to get confirmation.

Reply:
Ach I'm in two minds. I think your explanation makes sense, but I'm a little worried about merging this in during regression testing especially as it's not tested. How do you feel about this being pulled out as a separate change (we can potentially as an issue) so we can add tests around it and merge later?

Reply:
Fair enough. Let me take it out and the logging will help us either confirm that's the issue or discard that hypothesis. Then we can either add the change in a point release or in v1.30.